src/FffGcodeWriter.cpp

```diff
@@ -2529,98 +2529,7 @@ bool FffGcodeWriter::processInsets(
-                const int layer_nr = gcode_layer.getLayerNr();
-                auto filled_area_above_res = getOutlineOnLayer(part, layer_nr + roofing_layer_count);
-                if (! no_small_gaps_heuristic)
-                {
-                    for (int layer_nr_above = layer_nr + 1; layer_nr_above < layer_nr + roofing_layer_count; layer_nr_above++)
-                    {
-                        Polygons outlines_above = getOutlineOnLayer(part, layer_nr_above);
-                        filled_area_above_res = filled_area_above_res.intersection(outlines_above);
-                    }
-                }
-                if (layer_nr > 0)
-                {
-                    // if the skin has air below it then cutting it into regions could cause a region
-                    // to be wholely or partly above air and it may not be printable so restrict
-                    // the regions that have air above (the visible regions) to not include any area that
-                    // has air below (fixes https://github.com/Ultimaker/Cura/issues/2656)
-
-                    // set air_below to the skin area for the current layer that has air below it
-                    Polygons air_below = getOutlineOnLayer(part, layer_nr).difference(getOutlineOnLayer(part, layer_nr - 1));
-
-                    if (! air_below.empty())
-                    {
-                        // add the polygons that have air below to the no air above polygons
-                        filled_area_above_res = filled_area_above_res.unionPolygons(air_below);
-                    }
-                }
-
-                return filled_area_above_res;
-            }();
-
-            if (filled_area_above.empty())
-            {
-                return true;
-            }
-
-            const auto point_view = ranges::views::transform(
-                [](auto extrusion_junction)
-                {
-                    return extrusion_junction.p_;
-                });
-
-            for (const auto& path : part.wall_toolpaths)
-            {
-                for (const auto& wall : path)
-                {
-                    for (const auto& p : wall | point_view)
-                    {
-                        if (! filled_area_above.inside(p))
-                        {
-                            return true;
-                        }
-                    }
-
-                    for (const auto& window : wall | point_view | ranges::views::sliding(2))
-                    {
-                        auto p0 = window[0];
-                        auto p1 = window[1];
-                        if (PolygonUtils::polygonCollidesWithLineSegment(filled_area_above, p0, p1))
-                        {
-                            return true;
-                        }
-                    }
-                }
-            }
-            return false;
-        }();
-
+        auto use_roofing_config = false;
```

Comment:
This variable doesn't serve any purpose now

Reply:
We still want to implement something in place of the removed code; added a comment to reflect that d8cbf1f186a8d8142ad89d3a83495122bb4e9494